staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -749,6 +749,7 @@ func DefaultBuildHandlerChain(apiHandler http.Handler, c *Config) http.Handler {
 	handler = genericfilters.WithTimeoutForNonLongRunningRequests(handler, c.LongRunningFunc, c.RequestTimeout)
 	handler = genericfilters.WithWaitGroup(handler, c.LongRunningFunc, c.HandlerChainWaitGroup)
 	handler = genericapifilters.WithRequestInfo(handler, c.RequestInfoResolver)
+	handler = genericapifilters.WithTracing(handler)
```

Comment:
I expected to see a feature or configuration gate here so this filter was not active if feature was off or unconfigured

Reply:
Unless an "SDK" is registered (which is done in the InitTraces function in staging/src/k8s.io/component-base/traces/utils.go using `otel.SetTracerProvider(...)`), OpenTelemetry uses a NoOp tracer and doesn't propagate context.  This means the Tracing filter doesn't do anything unless we register an SDK, which is controlled by the feature gate.

Reply:
I'd still prefer to feature gate this.

Reply:
Also I don't think this should be relying on globals to be set up right elsewhere, I think a handle or context should be passed here.

Reply:
Commented above already: even when the feature gate is on by default in the future, the functionality should be disabled if there is no config provided. The current code has a default config file path and I assume will fail when the config does not exist. 

In other words: +1 for opt-in, plus even after feature gate goes away some day.

Reply:
This line is now feature gated.  The feature is disabled if the flag: `--tracing-config-file` is not provided.